src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/EETypeNode.cs

```diff
@@ -406,23 +406,33 @@ private void OutputBaseSize(ref ObjectDataBuilder objData)
 
         private void OutputRelatedType(NodeFactory factory, ref ObjectDataBuilder objData)
         {
-            TypeDesc relatedType = _type.BaseType;
+            ISymbolNode relatedTypeNode = null;
```

Comment:
Maybe the change in this file should go in a separate commit? It's useful, but unrelated and this commit is pretty big as it is...